pkg/cri/sbserver/events.go

```diff
@@ -393,7 +393,7 @@ func handleContainerExit(ctx context.Context, e *eventtypes.TaskExit, cntr conta
 		},
 	)
 	if err != nil {
-		if !errdefs.IsNotFound(err) {
+		if !errdefs.IsNotFound(err) && !errdefs.IsUnavailable(err) {
```

Comment:
No the unavailable error is independent of the sandbox implementation - it is a result of a failed grpc connection (since the shim is dead). The error would originate from 
https://github.com/containerd/containerd/blob/00666764b80c0eeb9f7da807b0bd0eac9c1a729a/runtime/v2/shim.go#L700

When we would try to contact the shim to get its state but realize we cant reach it



The error is something like below 
```log
ERRO[2023-09-19T16:11:06.940444000Z] StopContainer for "6668b4c6f405e58613b8cbfb1a5299b01d4eaf288115766a36f008156f746cad" failed  error="rpc error: code = Unavailable desc = failed to stop container \"6668b4c6f405e58613b8cbfb1a5299b01d4eaf288115766a36f008156f746cad\": connection error: desc = \"transport: Error while dialing: dial unix <path>: connect: connection refused\": unavailable"
```

Reply:
TIL. The grpc will reconnect to the tcp but fast-fail on unix socket. Thanks for details. I think we should have work item to setup the integration for the grpc shim. @adityaramani would you mind creating the issue to track this? Thanks

Reply:
@fuweid Here is the issue: https://github.com/containerd/containerd/issues/9137

Please tweak the details as you think is fit